src/App.jsx

```diff
@@ -14,33 +15,86 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const ABC = '1';
+const L = '2';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function Sort(arr, reverse, sort) {
+  let tempArr = [...arr];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  switch (sort) {
+    case ABC:
+      tempArr = tempArr.sort((a, b) => a.localeCompare(b));
+      break;
+    case L:
+      tempArr = tempArr.sort((a, b) => a.length - b.length);
+      break;
+    default:
+      break;
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (reverse) {
+    tempArr.reverse();
+  }
+
+  return tempArr;
+}
+
+export const App = () => {
+  const [isReverse, setIsReverse] = useState(false);
+  const [sort, setSort] = useState('');
+
+  const visibleGoods = Sort(goodsFromServer, isReverse, sort);
+
+  function Reset() {
+    setIsReverse(false);
+    setSort('');
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-danger ${sort === ABC ? null : 'is-light'}`}
```

Comment:
To set classes conditionally it's better to use [classnames](https://www.npmjs.com/package/classnames) or [clsx](https://www.npmjs.com/package/clsx) library